src/Twilio/Types.hs

```diff
@@ -104,7 +104,14 @@ makeTwilioDELETERequest :: Monad m
 makeTwilioDELETERequest resourceURL =
   makeTwilioRequest resourceURL <&> (\req -> req {
     method = "DELETE",
-    checkStatus = \status responseHeaders cookieJar -> case status of
-      Status 204 _ -> Nothing
-      _            -> Just . SomeException $ StatusCodeException status responseHeaders cookieJar
+    checkResponse = throwForNon204
   })
+
+
+throwForNon204 :: Request -> Response BodyReader -> IO ()
+throwForNon204 _ resp =
+  case responseStatus resp of
+    Status 204 _ -> return ()
+    status       -> do
```

Comment:
Since `status` is unused, why not

```hs
    Status 204 _ -> return ()
    _ -> do
```

Also, we may want to prefer `pure` to `return`, but it's a minor thing. Better to catch with some kind of style-check in the future...